priv/posts/performance-methodologies.md

```diff
@@ -0,0 +1,170 @@
+- **Interconnects**: CPU, memory, I/O
+
+The USE method is most effective for resources that suffer performance degradation under high utilization or saturation, leading to bottlenecks. Fortunately, they are not common system bottlenecks, as they are typically designed to provide an excess of throughput. Unfortunately, if they are the problem, can be difficult to solve.
+
+After you get the list of resources, try to create some metrics for it:
+
+![List of resources to create metric](/images/performance-analysis/list-resources-use.png)
+
+The process of elimination is good for us. Eliminate a possible resource bottleneck may help us to focus on another resource limiting our scope.
+
+### Drill Down
+
+The process can involve digging down through deeper layers of the software stack, to hardware if necessary, to find the root cause of the issue. Particularly, I try to apply this methodology whenever that I can; It's hard to apply it without a big picture of your software, but when you get more experience on it, should be easier.
+
+> Analysis (Monitoring) is the base of all! Without it, we can't fix any bug.
+
+Such deeper analysis may involve the creation of custom tools and inspection of source code (if available). Here is where most of the drilling takes place, peeling away layers of the software stack as necessary to find the root cause.
+
+For instance, let's imagine an application that after a month in production environment has begun to perform poorly.
+
+**Five Whys**
+
+1. A database has performing poorly for some queries. Why?
+2. It's delayed by disk I/O due to memory paging. Why?
+3. Database memory usage has grown too large. Why?
+4. The allocator is consuming more memory than expected. Why?
+5. The allocator has a memory fragmentation issue.
+
+This is a good real sample extracted from [System Performance](https://www.goodreads.com/book/show/18058001-systems-performance) book. There is not limit to go deep into _Why?_, but, one has to when the software is performing well.
+
+![Five why - Drill Down](/images/performance-analysis/5-whys.png)
+
+
+### Scientific Method
+
+The _scientific method_ studies the _unknown_ by making hypotheses and testing them. The `unknown` here can mean the `unknown-unknown` as discussed in [`Know-Unknows`](#know-unknowns) section.
+
+Every _scientific method_ consists:
+
+1. Formulation of a question?
+2. Hypothesis
+3. Prediction
+4. Testing
+5. Analysis
+
+> To futher information about how _scientific methods_ works, see [here](https://en.wikipedia.org/wiki/Scientific_method)
+
+First, you create a question based on performance problem, for instance: _Why my application have a degradation of throughput?_.
+After, you hypothesize what the cause of poor performance may be, _CPU Miss rate_. Then you construct a test to prove your theory, on this case I would recommend `Valgrind`.
+The results collected by _Testing_ step is analyzed and then you have a better idea of what's happening.
```

Comment:
```suggestion
Collect results from your previous step and analyze how it behaves over time. It will give you a better idea of what components are connected and ultimately affected.
```